spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala

```diff
@@ -0,0 +1,654 @@
+        } else if (ev == NumericDouble) {
+          tensorBuilder.setDatatype(DataType.DOUBLE)
+          tensor.storage().array().foreach(data => tensorBuilder.
+            addDoubleData(ev.toType[Float](data)))
+        }
+        tensor.size().foreach(size => tensorBuilder.addSize(size))
+        attributeBuilder.setTensorValue(tensorBuilder.build)
+      }
+    }
+
+  }
+
+/**
+ * DataConverter for [[com.intel.analytics.bigdl.nn.VariableFormat]]
+ */
+  object VariableFormatConverter extends DataConverter {
+
+    override def getAttributeValue[T: ClassTag](attribute: AttrValue)
+      (implicit ev: TensorNumeric[T]): AnyRef = {
+      val format = attribute.getVariableFormatValue
+      format match {
+        case VarFormat.DEFAULT => VariableFormat.Default
+        case VarFormat.ONE_D => VariableFormat.ONE_D
+        case VarFormat.IN_OUT => VariableFormat.IN_OUT
+        case VarFormat.OUT_IN => VariableFormat.OUT_IN
+        case VarFormat.IN_OUT_KW_KH => VariableFormat.IN_OUT_KW_KH
+        case VarFormat.OUT_IN_KW_KH => VariableFormat.OUT_IN_KW_KH
+        case VarFormat.GP_OUT_IN_KW_KH => VariableFormat.GP_OUT_IN_KW_KH
+        case VarFormat.GP_IN_OUT_KW_KH => VariableFormat.GP_IN_OUT_KW_KH
+        case VarFormat.OUT_IN_KT_KH_KW => VariableFormat.OUT_IN_KT_KH_KW
+        case VarFormat.EMPTY_FORMAT => null
+      }
+    }
+
+    override def setAttributeValue[T: ClassTag](attributeBuilder: AttrValue.Builder,
+      value: Any, valueType: universe.Type = null)(implicit ev: TensorNumeric[T]): Unit = {
+      attributeBuilder.setDataType(DataType.VARIABLE_FORMAT)
+      if (value != null) {
+        val format = value.asInstanceOf[VariableFormat]
+        val formatValue = format match {
+          case VariableFormat.Default => VarFormat.DEFAULT
+          case VariableFormat.ONE_D => VarFormat.ONE_D
+          case VariableFormat.IN_OUT => VarFormat.IN_OUT
+          case VariableFormat.OUT_IN => VarFormat.OUT_IN
+          case VariableFormat.IN_OUT_KW_KH => VarFormat.IN_OUT_KW_KH
+          case VariableFormat.OUT_IN_KW_KH => VarFormat.OUT_IN_KW_KH
+          case VariableFormat.GP_OUT_IN_KW_KH => VarFormat.GP_OUT_IN_KW_KH
+          case VariableFormat.GP_IN_OUT_KW_KH => VarFormat.GP_IN_OUT_KW_KH
+          case VariableFormat.OUT_IN_KT_KH_KW => VarFormat.OUT_IN_KT_KH_KW
+        }
+        attributeBuilder.setVariableFormatValue(formatValue)
+      } else {
+        attributeBuilder.setVariableFormatValue(VarFormat.EMPTY_FORMAT)
+      }
+   }
+}
+/**
+ * DataConverter for [[com.intel.analytics.bigdl.nn.InitializationMethod]]
+ */
+  object InitMethodConverter extends DataConverter {
+
+    override def getAttributeValue[T: ClassTag](attribute: AttrValue)
+      (implicit ev: TensorNumeric[T]): AnyRef = {
+      val initMemethod = attribute.getInitMethodValue
+      val initType = initMemethod.getMethodType
+      val methodData = initMemethod.getDataList
+      initType match {
```

Comment:
As discussed offline. it's not only for Enum type, it will be for all user defined class, I will add this functionality support in a separate PR